qt/scientific_interfaces/ISISReflectometry/GUI/Instrument/InstrumentPresenter.cpp

```diff
@@ -23,16 +24,33 @@ InstrumentPresenter::InstrumentPresenter(IInstrumentView *view,
                                          Instrument instrument)
     : m_view(view), m_model(std::move(instrument)) {
   m_view->subscribe(this);
+}
+
+void InstrumentPresenter::acceptMainPresenter(
```

Comment:
There are a few small chunks of repeated code across the presenters. Would it make sense to move these to a bare-bones class for the various presenters to inherit from? 
p.s. This may well break some best practice I'm not aware of
p.p.s. The total number of repeated lines are small, however, so creating a new class may result in more code

Reply:
I'd rather leave this for now, but I think it might be worth doing in a future PR. I might also be able to tidy it up so there's a single interface for all of the tab presenters, instead of a separate interface for each one. I've added a note to #23051 to look into this.